app/assets/javascripts/discourse/app/controllers/create-invite.js

```diff
@@ -0,0 +1,147 @@
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import { getAbsoluteURL } from "discourse-common/lib/get-url";
+import discourseComputed from "discourse-common/utils/decorators";
+import { ajax } from "discourse/lib/ajax";
+import { extractError } from "discourse/lib/ajax-error";
+import copyText from "discourse/lib/copy-text";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import Group from "discourse/models/group";
+import I18n from "I18n";
+
+export default Controller.extend(ModalFunctionality, {
+  onShow() {
+    let inviteKey = "";
+    for (let i = 0; i < 32; ++i) {
+      inviteKey += "0123456789abcdef"[Math.floor(Math.random() * 16)];
+    }
+
+    this.setProperties({
```

Comment:
I might be fooled by diff view, but we should define all these values in top level too.